tests/simulation/test_simulation_device.cpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <random>
+#include "device_fixture.hpp"
+
+void generate_data(std::vector<uint32_t> &data, uint32_t size_in_bytes){
```

Comment:
why not have the function signature be 

`std::vector<uint32_t> data generate_data( uint32_t size_in_bytes )` 

?

Reply:
RVO would prevent copying if it was `std::vector<uint32_t> wdata = generate_data(size_in_bytes)` right

Reply:
Function signature changed to `std::vector<uint32_t> generate_data( uint32_t size_in_bytes )`

Reply:
> RVO would prevent copying if it was `std::vector<uint32_t> wdata = generate_data(size_in_bytes)` right

yes it's a very simple function so RVO should kick in